app/extensions/api/namespace.py

```diff
@@ -83,6 +83,8 @@ def decorator(func):
             from app.extensions import oauth2
             from app.modules.users import permissions
 
+            self.register_decorator_in_function(func, decorator)
```

Comment:
Please, use `oauth2.require_oauth(*oauth_scopes)` here (and move it closer to the `oauth_protected_func = ...` line) since otherwise this will call `self.doc` and other unnecessary stuff.